code/game/objects/items/robot/items/tools.dm

```diff
@@ -225,8 +240,7 @@
 	//set the reference & update icons
 	for(var/obj/item/tool as anything in omni_toolkit)
 		if(initial(tool.name) == toolkit_menu)
-			reference = tool
-			tool_behaviour = initial(tool.tool_behaviour)
+			set_internal_tool(tool)
```

Comment:
not commenting on anything else right now, just that it feels kinda awkward to loop over `omni_toolkit` twice
```dm
for(var/obj/item/tool as anything in omni_toolkit)
		if(initial(tool.name) == toolkit_menu)
				set_internal_tool(tool)
```
because `set_internal_tool(tool)` then also does its own such loop
```dm
for(var/obj/item/internal_tool as anything in omni_toolkit)
		if(internal_tool == tool)
				(...)
```